### PR TITLE
Alter LauchImage logo during the 2024 Chistmas holidays

### PR DESCRIPTION
**Pull request by daschuer**
This is a stub for the 2024 Chistmas holidays launch image. 

Now we can become creative !

mixxx-icon-logo-christmas

### CODE REVIEW

**Comment by daschuer** [comment]
Yes of cause. But how about the image itself? My is only a placeholder.

**Comment by acolombier** [comment]
I **love** the idea! I'll submit a visual in a bit to help the initiative, but obviously open to any which we like!

**Comment by JoergAtGithub** [comment]
I like the monocrome look of the Mixxx icon - also in combination with the Mixxx logo next to it.

File: src/skin/legacy/launchimage.cpp [hunk 1]
```diff
@@ -9,9 +10,39 @@
 
 #include "moc_launchimage.cpp"
 
+namespace {
+bool isIn2024ChristmasHollidays() {
+    auto currentDate = QDate::currentDate();
+    return currentDate >= QDate(2024, 12, 24) &&
+            currentDate <= QDate(2025, 1, 1);
```
Comment (acolombier) [inline src/skin/legacy/launchimage.cpp:17]: Shall we extend that to the 7th, just to make it last a little bit longer?

**Comment by Eve00000** [comment]
ho ho how nice ;-)
normally the christmas-periode stops after epiphany (6/1, 3 kings: children singing at the door), the trees get burned and the decoration should be put away.

**Comment by daschuer** [comment]
The 6th works for me as well as the latest date. I have original proposed the 1st to not annoy anyone. 

The santa head looks a bit like a VLC copy ... is this an issue? 
grafik

**Comment by daschuer** [comment]
> Edit: Only done the main logo for now. Thoughts?

Thanks, the Launch image logo is already enough IMHO. We need to hit the small path between nice and tacky. 

@acolombier Do you have interest to take this over and style one incorporating the suggestions?

**Comment by JoergAtGithub** [comment]
This PR is missing the logo file itself

**Comment by daschuer** [comment]
> I don't think I have enough expertise in Inscape top get something fully vectorial.
Are we okay with that?

It just should not look bad one a High-DPI device.

**Comment by ywwg** [comment]
> The santa head looks a bit like a VLC copy ... is this an issue?

no it's definitely a different image and they don't own the idea of a santa hat on a logo

**Comment by acolombier** [comment]
Sorry @daschuer , I didn't mean to push to your branch directly... Let me know if you would want me to move that to a separate PR, as originally plan. Here is the current look - it should support a pretty High DPI before looking pixelated, but keen to hear your tests

image

File: src/skin/legacy/launchimage.cpp [hunk 2]
```diff
@@ -9,9 +10,39 @@
 
 #include "moc_launchimage.cpp"
 
+namespace {
+bool isIn2024ChristmasHolidays() {
+    auto currentDate = QDate::currentDate();
+    return currentDate >= QDate(2024, 12, 24) &&
+            currentDate <= QDate(2025, 1, 6);
```
[the remark this answers is not in the document]
Comment (fwcd) [reply src/skin/legacy/launchimage.cpp:17]: It's a nice easter egg if we keep it

**Comment by ronso0** [comment]
> image

If the santa cap route is okay for everyone, it'd look better if the logo base color is light grey so the snow contrast is higher.

**Comment by fwcd** [comment]
Personally I actually preferred the almost white one earlier, IMO it makes the snow a bit more of a subtle detail and doesn't change the logo that much. With the darker ones I find the contrast between the text and the progress bar to be a bit odd, where the snow is emphasized a lot more than the logo itself.

**Comment by JoergAtGithub** [comment]
Could you try to scale down the icon to the height of the letter M ?

**Comment by acolombier** [comment]
Here is the result with scaling and `#bbb` text colour
image

**Comment by acolombier** [comment]
Shall I run a quick poll on Zulip? Or do we have consensus on that last version?

**Comment by daschuer** [comment]
There are some white border pixel shining though, else it looks good in 200 %: 
grafik
Is the santa had available under a compatible license?

**Comment by acolombier** [comment]
> There are some white border pixel shining though

Will try to remove them

> Is the santa had available under a compatible license?

Meh actually it might not be. Looking at the license, it seems that we can use it and edit it for free, but we must credit the author ([sourcehttps://www.freepik.com/free-vector/christmas-hat-design_137585811.htm#fromView=keyword&page=1&position=2&uuid=6adf5f02-ce77-4089-af02-445058bae8a6))
Shall we a quick mention in the about page?

**Comment by ywwg** [comment]
these all look great

**Comment by acolombier** [comment]
FWI, I'm in touch with a Mixxx user who wants to contribute and is redrawing the assets. I'll push that tonight or tomorrow!

**Comment by acolombier** [comment]
I have pushed the new logo, with the Christmas assets redrawn by @peineduper!

image

Good news - this is now fully vectorial so it should look great on High DPI @daschuer

**Comment by ronso0** [comment]
Could you please temporarily adjust the date range so we can test it now?
(commit as --fixup will prevent this being merged)

**Comment by Holzhaus** [comment]
I guess you can just change your system time ;)

**Comment by ronso0** [comment]
Umm, anyone wanting to try this has to change the system time (and rememeber to revert it)?

**Comment by ywwg** [comment]
changed my system clock around, it works

**Comment by acolombier** [comment]
The latest push also enable the Christmas spash permanently to help testing!

**Review by Swiftb0y — APPROVED** [review]
LGTM. please remove the last commit and squash the fixups.

**Comment by fwcd** [comment]
Unrelated, any chance we could get a 2.5 merge (potentially after this) soon? Would unblock a few PRs of mine 😄

**Comment by acolombier** [comment]
Sorry guys, I'm AFK till late tonight (GMT) so might not have time to do so till tomorrow. Please feel free to take the lead on this and do the fixup!

**Comment by daschuer** [comment]
@peineduper Thank you for you beautiful logo mod. Can you please grand us also formally the permission to distribute it with Mixxx? Please sign https://docs.google.com/a/mixxx.org/spreadsheet/viewform?formkey=dEpYN2NkVEFnWWQzbkFfM0ZYYUZ5X2c6MQ to do that and comment here when done. Thank you.

**Comment by peineduper** [comment]
Hey @daschuer, thank you ! 
Commenting to let you know that I just signed the agreement.

**Comment by acolombier** [comment]
Squashed the fixup and drop the testing commit, so all ready to go!

**Comment by daschuer** [comment]
@peineduper Thank you, how do you like to appear in the contributor list in the about box?

**Comment by daschuer** [comment]
@acolombier did you consider to adjust the progress bar color to a darker gray?

**Comment by peineduper** [comment]
> @peineduper Thank you, how do you like to appear in the contributor list in the about box?

"Emilien Colombier" would be great ! Thank you

**Comment by acolombier** [comment]
> did you consider to adjust the progress bar color to a darker gray?

I tried different tones, but in the end I though the white gave the best results. Feel free to adjust it tho!

**Comment by JoergAtGithub** [comment]
Looks great! Thank you!

**Comment by m0dB** [comment]
I just started up Mixxx and even though I did read this PR while it was in progress, the logo change took me by surprise. This was a great idea and very nicely executed! Merry Mixxxmas everyone!

**Pull request merged**